packages/server/test/integration/video_capture_spec.ts

```diff
@@ -108,4 +90,45 @@ describe('Video Capture', () => {
       expect(_pt.write).calledOnce
     })
   })
+
+  context('#start.endVideoCapture', () => {
+    let tmpFilename
+
+    beforeEach(() => {
+      tmpFilename = path.join(fse.mkdtempSync(path.join(os.tmpdir(), 'cy-video-')), 'video.mp4')
+    })
+
+    it('ends immediately if more than two frames written', async () => {
+      const { writeVideoFrame, endVideoCapture } = await startSpiedVideoCapture(tmpFilename)
+
+      writeVideoFrame(fse.readFileSync(image1Path))
+      writeVideoFrame(fse.readFileSync(image2Path))
+      writeVideoFrame(fse.readFileSync(image3Path))
+
+      await endVideoCapture(false)
```

Comment:
nit: mostly to keep the same readability style
```suggestion
      const waitForMoreFrames = false
      await endVideoCapture(waitForMoreFrames)
```